File: app/src/main/java/com/itverse/futuris/data/daos/ComposantDao.java
package com.itverse.futuris.data.daos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.itverse.futuris.data.entities.Composant;

import java.util.List;

@Dao
public interface ComposantDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertComposant(Composant composant);
    @Delete
    void deleteComposant(Composant composant);
    @Query("SELECT * FROM composant_table WHERE composant_name = :composantName")
    LiveData<Composant> getComposant(String composantName);
    @Query("SELECT * FROM composant_table")
    LiveData<List<Composant>> getAllComposant();
}

File: app/src/main/java/com/itverse/futuris/data/daos/ComposantDao.kt
package com.itverse.futuris.data.daos

import androidx.room.*
import com.itverse.futuris.data.entities.Composant
import kotlinx.coroutines.flow.Flow

@Dao
interface ComposantDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(composant: Composant): Long

    @Delete
    suspend fun delete(composant: Composant)

    @Transaction
    @Query("SELECT * FROM composant_table WHERE projectId = :projectId")
    fun getAllComposantsFromProject(projectId: Long): Flow<List<Composant>>

    @Transaction
    @Query("SELECT * FROM composant_table WHERE projectId = :projectId")
    suspend fun getAllComposantsFromProject_(projectId: Long): List<Composant>

}